zebrad/src/components/mempool/storage.rs

```diff
@@ -0,0 +1,105 @@
+use std::collections::{HashMap, HashSet, VecDeque};
+
+use zebra_chain::{
+    block,
+    transaction::{UnminedTx, UnminedTxId},
+};
+use zebra_consensus::error::TransactionError;
+
+use super::MempoolError;
+
+#[cfg(test)]
+pub mod tests;
+
+const MEMPOOL_SIZE: usize = 2;
+
+#[allow(dead_code)]
+#[derive(Clone, Debug)]
+pub enum State {
+    /// Rejected because verification failed.
+    Invalid(TransactionError),
+    /// An otherwise valid mempool transaction was mined into a block, therefore
+    /// no longer belongs in the mempool.
+    Confirmed(block::Hash),
+    /// Stayed in mempool for too long without being mined.
+    // TODO(2021-08-20): set expiration at 2 weeks? This is what Bitcoin does.
+    Expired,
+    /// Transaction fee is too low for the current mempool state.
+    LowFee,
+    /// Otherwise valid transaction removed from mempool, say because of FIFO
+    /// (first in, first out) policy.
+    Victim,
+}
+
+#[derive(Clone, Default)]
+pub struct Storage {
+    /// The set of verified transactions in the mempool. Currently this is a
+    /// cache of size 2.
```

Comment:
It would be handy to just change this in one place:
```suggestion
    /// The set of verified transactions in the mempool. This is a
    /// cache of size [`MEMPOOL_SIZE`].
```